packages/autocomplete-js/src/__tests__/requester.test.ts

```diff
@@ -346,6 +350,93 @@ describe('requester', () => {
     });
   });
 
+  test('batches calls when possible and re-dispatches results to the right sources across requester instances', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+
+    const searchClient = createSearchClient({
+      search: jest.fn(() =>
+        Promise.resolve(
+          createMultiSearchResponse<{ label: string }>(
+            { hits: [{ objectID: '1', label: 'Hit 1' }] },
+            { hits: [{ objectID: '2', label: 'Hit 2' }] }
+          )
+        )
+      ),
+    });
+
+    const getResults1 = (params) =>
+      createRequester(fetchAlgoliaResults)({
+        transformResponse: (response) => response.hits,
+      })(params);
+
+    const getResults2 = (params) =>
+      createRequester(fetchAlgoliaResults)({
+        transformResponse: (response) => response.hits,
+      })(params);
+
+    autocomplete({
+      container,
+      panelContainer,
+      getSources({ query }) {
+        return [
+          {
+            sourceId: 'hits',
+            getItems() {
+              return getResults1({
+                searchClient,
+                queries: [
+                  {
+                    indexName: 'indexName',
+                    query,
+                  },
+                ],
+              });
+            },
+            templates: {
+              item({ item }) {
+                return JSON.stringify(item);
+              },
+            },
+          },
+          {
+            sourceId: 'other-hits',
+            getItems() {
+              return getResults2({
+                searchClient,
+                queries: [
+                  {
+                    indexName: 'indexName',
+                    query,
+                  },
+                ],
+              });
+            },
+            templates: {
+              item({ item }) {
+                return JSON.stringify(item);
+              },
+            },
+          },
+        ];
+      },
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, { target: { value: 'a' } });
+
+    await waitFor(() => {
+      expect(
+        panelContainer.querySelector<HTMLElement>('.aa-Panel')
+      ).toBeInTheDocument();
+
+      expect(searchClient.search).toHaveBeenCalledTimes(1);
```

Comment:
nice test!

Reply:
Agree on the end-to-end test. I believe this was initially in the PR then removed?

I recall our end-to-end setup not to be optimal, but this is a good opportunity to fix it given that one underlying issue is how we expect all our builds to work similarly (and in this case, they don't).

Reply:
there was an UMD E2E test initially indeed, but it wasn't actually easily testing for the case we were fixing. I'll recreate it in a separate PR

Reply:
For the record, FX-1228 in the backlog is about this.